### ISSUE TITLE
No sensors in entities

### ISSUE THREAD

**Issue by alexdelprete** [LordMike/MBW.BlueRiiot2MQTT#45]
Hi Mike,

as a follow-up to my post on HA forum, I wanted to know how to fix the issue in subject.

The addon seems to be working fine, I can see data in MQTT, but I can't find the sensors in the entities list, I looked for pool name, for blueriiot and bluconnect, and I also checked the created device Blueriiot2MQTT, but I only see 3 disabled entities under it:

![image](https://user-images.githubusercontent.com/7027842/115128075-92886a80-9fdb-11eb-85ea-1db8958f155c.png)

I don't know if there's a log somewhere I can check or a debug mode...tell me if I need to provide you some in-depth info/data.

Thanks for the help.

Alessandro

**Comment by LordMike** [comment]
If you check all entities from the MQTT integration in HASS, do you find it then?

#38 seems identical, where the issue was what I recommended: search for the pool name.

You can also check the log on the Blueriiot2Mqtt container, and see if it reports anything like `New data ready for pool` or `Pushed N discovery documents, ..` (where N is >4 or so..)

**Comment by alexdelprete** [comment]
Under the MQTT integration I see no entities.

![image](https://user-images.githubusercontent.com/7027842/115152237-99ad8800-a070-11eb-9f91-10dc84c025b5.png)

Here's the log of the addon:

```
[15:59:13 INF] Loading extra config file at /data/options.json
[2021-04-18 15:59:14+00:00 INF] [MqttNet.ManagedMqttClient] Started
[2021-04-18 15:59:14+00:00 INF] [MqttNet.MqttClient] Connected.
[2021-04-18 15:59:14+00:00 INF] [Microsoft.Hosting.Lifetime] Application started. Press Ctrl+C to shut down.
[2021-04-18 15:59:14+00:00 INF] [Microsoft.Hosting.Lifetime] Hosting environment: Production
[2021-04-18 15:59:14+00:00 INF] [Microsoft.Hosting.Lifetime] Content root path: /app
[2021-04-18 15:59:16+00:00 INF] [MBW.BlueRiiot2MQTT.Service.BlueRiiotMqttService] Discovered new pool, 'Piscina Casa Fiano' (eb743aeb-f698-4383-bacb-57b510aa773e)
[2021-04-18 15:59:18+00:00 INF] [MBW.BlueRiiot2MQTT.Service.PoolUpdater.SingleBlueRiiotPoolUpdater] New data ready for pool 'Piscina Casa Fiano' at 04/18/2021 16:54:00 (interval 01:12:00). Waiting 00:55:20.4402174
[2021-04-18 15:59:18+00:00 INF] [MBW.HassMQTT.HassMqttManager] Pushed 20 discovery documents, 19 values and 20 attribute changes
```

I looked for "Piscina Casa Fiano" also 2 days ago, there are no entities that contain any of those 3 words.

Are those autocreated entities associated to the MQTT integration? If so, I have none, so that's the problem I guess.

Last thing: through MQTT Explorer, I can see all data correctly created and updated in the MQTT server. Here's a screenshot:

![image](https://user-images.githubusercontent.com/7027842/115152376-21939200-a071-11eb-8862-ab2e22adb541.png)

**Comment by LordMike** [comment]
Huh, that's odd.. your name has spaces in it ... that's all I can think of .. BR2MQTT is correctly pushing discovery documents as you see in MQTT. Could you check HASS' logs to see if there are any warnings or errors about invalid discovery documents?

**Comment by alexdelprete** [comment]
The only thing I found in HA logs is this;

![image](https://user-images.githubusercontent.com/7027842/115160404-43ecd600-a098-11eb-908b-a1e6e876c27c.png)

I also renamed the pool, now it's `HomePool`, without spaces. Waited some time. No entities with `HomePool` or substring home/pool in the name.

![image](https://user-images.githubusercontent.com/7027842/115160465-88787180-a098-11eb-8c11-5388cbcdc46f.png)

**Comment by LordMike** [comment]
I'm still at a loss. 

What if you enable debug logging on HASS for MQTT discovery?

Call this service in HASS (use the debug panel):

```
service: logger.set_level
data:
  homeassistant.components.mqtt: debug
```

Then view the logs in either the log viewer or the console output (if its running docker f.ex.). See if HASS receives the discovery documents, and if it logs anything out of the ordinary, like errors in the documents.

**Comment by LordMike** [comment]
I'm not used to running HassIO/HASS OS.. do you have an integration for MQTT that isn't the add-on, that shows the number of entities?

I f.ex. have this on `/config/integrations`:

![image](https://user-images.githubusercontent.com/1027111/115199608-ade69900-a0f3-11eb-8ca6-1aefd1737447.png)

I can see you have _some_ devices, so the MQTT integration _is_ working... .... the error you get seems to be an internal issue, perhaps related to MQTT. Maybe something is wrong with your HASS installation? I've tried looking up the error, and it shouldn't happen :|

**Comment by alexdelprete** [comment]
Like I wrote above, this is what I see under the MQTT integration box (no devices/entities):

![image](https://user-images.githubusercontent.com/7027842/115152237-99ad8800-a070-11eb-9f91-10dc84c025b5.png)

I suspect something is wrong for the MQTT integration, but I can't understand what it is.

Did you configure something in the configuration.yaml file for MQTT? Can you show it to me so I can check? Thanks.

**Comment by alexdelprete** [comment]
Fixed. The integration was installed months ago, and it pointed to core-mosquitto instance. When I installed your addon, I first installed the Mosquitto broker addon, but the integration was still pointing to core-mosquitto. :)

![image](https://user-images.githubusercontent.com/7027842/115248380-ebfdb000-a127-11eb-96f7-398564a765fc.png)
![image](https://user-images.githubusercontent.com/7027842/115248391-ee600a00-a127-11eb-9eaf-b971d4290291.png)
![image](https://user-images.githubusercontent.com/7027842/115248394-ef913700-a127-11eb-982e-666d3ac628a9.png)
